src/Common/RadixSort.h

```diff
@@ -252,7 +252,7 @@ struct RadixSort
         /// There are loops of NUM_PASSES. It is very important that they are unfolded at compile-time.
 
         /// For each of the NUM_PASSES bit ranges of the key, consider how many times each value of this bit range met.
-        CountType histograms[HISTOGRAM_SIZE * NUM_PASSES] = {0};
+        std::unique_ptr<CountType[]> histograms{new CountType[HISTOGRAM_SIZE * NUM_PASSES]{}};
```

Comment:
IMHO `std::vector` would look better in that case.

Reply:
Yes, but it can be (at least theoretically) worse, because it contains size and capacity fields that are unneeded. Also accessing data can involve offset in struct that will use more complex addressing mode... Though I'm almost sure that there will be no difference in practice.